myia/grad.py

```diff
@@ -0,0 +1,284 @@
+
+from typing import Set
+
+from .anf_ir import Apply, Constant, Parameter, Graph, ANFNode
+from . import primops
+from collections import defaultdict
+from .anf_ir_utils import dfs
+from .cconv import NestingAnalyzer
+
+
+def is_constant(x):
+    return isinstance(x, Constant)
+
+
+def is_graph(x):
```

Comment:
These two helper functions should probably go into `anf_ir_utils` (and `is_graph` should probably be `is_graph_node`, can get a bit confusing otherwise).

Reply:
Yeah, some cleanup's definitely needed here.